nilearn/input_data/nifti_masker.py

```diff
@@ -186,7 +182,7 @@ def fit(self, imgs=None, y=None):
         # Resampling: allows the user to change the affine, the shape or both
         if self.verbose > 0:
             print("[%s.fit] Resampling mask" % self.__class__.__name__)
-        self.mask_img_ = self._cache(image.resample_img, func_memory_level=1)(
+        self.mask_img_ = self._cache(image.resample_img, 1)(
```

Comment:
Each function has a reference memory_level (func_memory_level). If the user memory_level if equal or greater to func_memory_level, then the function is cached. The goal is, for example, to have several layers of caching. In NiftiMasker:
* level 1 only cache the output (masked data)
* level 2 will cache intermediate step of the pretreatment.